File: microservices/PassengerService/src/main/java/com/modsen/taxi/passengerservice/service/PassengerService.java
package com.modsen.taxi.passengerservice.service;

import com.modsen.taxi.passengerservice.dto.PassengerRequest;
import com.modsen.taxi.passengerservice.dto.PassengerResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import reactor.core.publisher.Mono;

public interface PassengerService {
    Mono<PassengerResponse> createPassenger(PassengerRequest passengerRequest);

    Mono<PassengerResponse> updatePassenger(Long id, PassengerRequest passengerRequest);

    Mono<PassengerResponse> getPassengerById(Long id);

    Mono<Page<PassengerResponse>> getAllPassengers(Pageable pageable, String firstName, String lastName, String email, boolean isActive);

    Mono<Void> deletePassenger(Long id);
}

File: microservices/PassengerService/src/main/java/com/modsen/taxi/passengerservice/service/PassengerService.kt
package com.modsen.taxi.passengerservice.service

import com.modsen.taxi.passengerservice.dto.PassengerRequest
import com.modsen.taxi.passengerservice.dto.PassengerResponse
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import reactor.core.publisher.Mono

interface PassengerService {
    fun createPassenger(passengerRequest: PassengerRequest): Mono<PassengerResponse>

    fun updatePassenger(
        id: Long,
        passengerUpdateRequest: PassengerRequest,
        principalEmail: String,
        isAdmin: Boolean
    ): Mono<PassengerResponse>

    fun getPassengerById(id: Long, principalEmail: String, isAdmin: Boolean): Mono<PassengerResponse>

    fun getAllPassengers(
        pageable: Pageable,
        firstName: String?,
        lastName: String?,
        email: String?,
        isActive: Boolean
    ): Mono<Page<PassengerResponse>>

    fun deletePassenger(id: Long, principalEmail: String, isAdmin: Boolean): Mono<Void>
}
